Tweet images instead of text

Ares358 opened the pull request:
Solves #90 

Changes made to get_tweet.py :
-updated get_tweet() so that it returns a random tweet each time it is called from within a loop. Before that it returned the same tweet if called from another script.

Changes made to generateImage.py :
-it now takes a random tweet from get_tweet.py and adds it on the images.
-default x and y co-ordinates changed so that the tweet appears in center of the image.
-could be imported into another script and used to generate an array of images via the "run(x)" function which generates x images with random tweets.

Changes made to tweet.py:
-The program uploads an image instead of the tweet. Another parameter could be passed to upload image WITH a tweet.

NOTE: It assumes that the generated images have the naming convention "post_{no}.png" and are all located in the same directory as the tweet.py file. It also assumes that there exists atleast 1 image with the above mentioned naming convention in the directory. If not, the script will enter an infinite loop when get_img_random() is called.

Ares358 committed:
Updated get_tweet() 

updated get_tweet() so that it returns a random tweet each time it is called from within a loop.

Ares358 committed:
Update generateImagePost.py

The program now gets a random tweet from get_tweet.py and prints it on a random image from img_bg.
The function "run" takes "x" as a parameter and repeats the above mentioned process for x no of times.

Ares358 committed:
Updated Tweet.py to post image instead of tweet

The program uploads an image instead of the tweet. Another parameter could be passed to upload image WITH a tweet.
It assumes that the generated images have the naming convention "post_{no}.png" and are all located in the same directory as the tweet.py file.
It also assumes that there exists atleast 1 image with the above mentioned naming convention in the directory.

Ares358 committed:
Add files via upload

Ares358 committed:
Update tweet.py

bobsany16 commented on tweet.py:
```diff
@@ -19,6 +21,16 @@
 access_token_secret = environ['access_token_secret']
 
 
+def get_img_random():
+    no=0
+    while(True):
+        no = random.randint(0,99)
+        if(os.path.exists('post_'+str(no)+'.png')):
+            return 'post_'+str(no)+'.png'
+        else:
+            continue
```
This is a nice approach. Trying to decide what's the best approach to error out when there's no image file just so it doesn't get stuck in an infinite loop... 

Could we keep track of the number of times it enters the else clause so that when it reaches a threshold (whatever we want), it breaks out of the while loop ? Do you think that's feasible ? 

Happy to brainstorm with you, if you'd like 😁

Ares358 replied:
Indeed. The two approaches we can take are -
1) Count the no. of times we enter the else clause and break the loop when a threshold has been reached. (As you mentioned above)
2) A linear search for files numbered 1 to 100 via os.path.exists and a flag variable.

The latter approach seems more reasonable to me due to the assurance that we'll either have no file or some file but it's also more resource taxing, in case we increase the no of images used in future.

bobsany16 replied:
I agree with you on the latter approach! It gives more assurance, whereas in my solution, we run the risk of not giving it enough time before finding the image! However, it can take quite an amount of resource if we do have more images in the future.

Regarding this concern, could we: 

1. Just extract all the file names from the image folder into an array in the .py file (chances are we'd have to loop through and then it won't be much different than a linear search) 
2. We perform whatever search to find the name based on the random function we already have
3. Or we can just do `random.choice(img_array)` to randomly do it from there. 

I think this is less resource taxing but could be slower if we have to fetch all the names of images every time we call the function.

Ares358 replied:
Made a linear scan of appropriate filenames into an array and a `random.choice(arr)` to select and post an image.
I'm calling the scan function only once each time the `run()` function is called and it then tweets for `len(arr)` no of times. We assume that the files won't be deleted by some other process/user during the runtime of the `for i in range(len(arr)):` loop to avoid performing linear search after every tweet.

bobsany16 replied:
Awesome! I'll take a look soon. Thank you thank you

bobsany16 replied:
I looked at the image folders and saw that the file name conventions have 'a' after some of them (16, 16a, etc). Could you incorporate this in the condition while searching for file names, when you have time ?

Ares358 replied:
Done. 
Not sure what the difference between the two types of files is though yet.
Awaiting some explanation in #90 .

bobsany16 replied:
Thank you so much !

VickiLanger replied:
There's nothing different between the files with different naming conventions. I just failed to keep the same convention

Ares358 committed:
Updated tweet.py

"No image exists in working directory" and infinite loop issue resolved. 

The program now scans and initializes the array "arr" containing image names and tweets every hr with 1 image from the array.
get_img_random() is obsolete now, hence is commented out.

Ares358 committed:
Updated tweet.py

updated to include both, post_{no} and post_{no+'a'} files into the array `arr` containing filenames

VickiLanger committed:
Merge branch 'production' into local

VickiLanger reviewed (approved):
Thanks for your contribution. I'm excited to see this in production

---
*Don't forget to follow the bot on [Twitter](https://www.twitter.com/LGBTQotd) to see the hard work you put in.*

Ares358 commented on tweet.py:
```diff
@@ -1,95 +1,99 @@
+    print('Used up all the images scanned. Generate more images and recall the function/program')
+
+
+run()
+
+if __name__ == "__main__":
+    main()
```
These 4 lines should be replaced with these 2 lines

`if __name__ == "__main__":`
    `run()`

`run()` already calls for `main()` at line 88

VickiLanger replied:
Done. Thanks!
